### PR TITLE
WA-1937: support transaction status of cancelled and rejected.

### PR DESCRIPTION
**Pull request by hailefrollo**

### CODE REVIEW

File: Sources/FrolloSDK/Model/Core Data/Aggregation/Transaction+CoreDataClass.swift [hunk 1]
```diff
@@ -62,6 +62,41 @@ public class Transaction: NSManagedObject, UniqueManagedObject {
         /// Scheduled. Transaction is scheduled for the future
         case scheduled
         
+        /// Cancelled. Transaction is cancelled
+        case cancelled
+        
+        /// Cancelled. Transaction is rejected
```
Comment (sincychacko-frollo) [inline Sources/FrolloSDK/Model/Core Data/Aggregation/Transaction+CoreDataClass.swift:68]: The comment should be rejected?

Comment (hailefrollo) [reply Sources/FrolloSDK/Model/Core Data/Aggregation/Transaction+CoreDataClass.swift:68]: Sorry, my mistake.

File: Sources/FrolloSDK/Model/Core Data/Aggregation/Transaction+CoreDataClass.swift [hunk 2]
```diff
@@ -62,6 +62,41 @@ public class Transaction: NSManagedObject, UniqueManagedObject {
...
+        case rejected
+        
+        public var rawValue: String {
+            switch self {
+                case .pending:
+                    return "0-pending"
```
Comment (sincychacko-frollo) [inline Sources/FrolloSDK/Model/Core Data/Aggregation/Transaction+CoreDataClass.swift:74]: Maybe we can have something like display order?

File: Sources/FrolloSDK/Resources/Database/FrolloSDKDataModel.xcdatamodeld/FrolloSDKDataModel-1.7.7.xcdatamodel/contents [hunk 3]
```diff
@@ -682,8 +685,8 @@
...
         <element name="ReportAccountBalance" positionX="-360" positionY="180" width="128" height="119"/>
         <element name="Tag" positionX="283.4609375" positionY="696.9375" width="128" height="89"/>
-        <element name="Transaction" positionX="277.71484375" positionY="20.28125" width="128" height="599"/>
+        <element name="Transaction" positionX="277.71484375" positionY="20.28125" width="128" height="629"/>
         <element name="TransactionCategory" positionX="507.08203125" positionY="227.6484375" width="128" height="164"/>
         <element name="User" positionX="-583.6328125" positionY="8.46484375" width="128" height="479"/>
     </elements>
-</model>
\ No newline at end of file
+</model>
```
Comment (sincychacko-frollo) [inline Sources/FrolloSDK/Resources/Database/FrolloSDKDataModel.xcdatamodeld/FrolloSDKDataModel-1.7.7.xcdatamodel/contents:692]: You might want to regenerate the test dbs.

File: Sources/FrolloSDK/Model/Core Data/Aggregation/Transaction+CoreDataClass.swift (hunk 2 above)
Comment (dipeshfrollo) [reply Sources/FrolloSDK/Model/Core Data/Aggregation/Transaction+CoreDataClass.swift:74]: Yeah maybe something like?

```
 public var rawValue: Int {
            switch self {
                case .pending:
                    return 0
                    .....
            }
}
```

**Review by dipeshfrollo — APPROVED** [review]

**Pull request merged**